actions/lib/action.py

```diff
@@ -10,10 +10,29 @@ def __init__(self, config):
 
     def _get_client(self):
         url = self.config['url']
-        token = self.config['token']
         verify = self._get_verify()
 
+        auth_method = self.config.get("auth_method", "token")
+        token = self.config["token"]
+
+        # token is passed during client init to allow client to also
+        # get the token from VAULT_TOKEN env var or ~/.vault-token
         client = hvac.Client(url=url, token=token, verify=verify)
+
+        # token is handled during client init
+        # other auth methods will override it as needed
+        if auth_method == "token":
+            pass
+
+        # where implemented prefer: client.auth.<method>.login
+        # as many client.auth_* methods are deprecated.
+
+        elif auth_method == "approle":
+            client.auth.approle.login(
+                role_id=self.config["role_id"],
+                secret_id=self.config["secret_id"],
+            )
```

Comment:
Probably good for a warning here if the auth_method doesn't exist.

Reply:
I went with a `NotImplementedError`. You should only get this if you're manually interacting with this python code somehow. StackStorm itself enforces that auth_method is one of `token` or `approle` with the enum in action metadata.